exchanges/coinbene/coinbene_test.go

```diff
@@ -0,0 +1,184 @@
+package coinbene
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/thrasher-corp/gocryptotrader/config"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+)
+
+// Please supply your own keys here for due diligence testing
+const (
+	testAPIKey              = ""
+	testAPISecret           = ""
+	canManipulateRealOrders = false
+
+	btcusdt = "BTC/USDT"
+)
+
+var c Coinbene
+
+func TestMain(m *testing.M) {
+	c.SetDefaults()
+	cfg := config.GetConfig()
+	err := cfg.LoadConfig("../../testdata/configtest.json")
+	if err != nil {
+		log.Fatalf("Test Failed - Coinbene Setup() init error:, %v", err)
+	}
+	coinbeneConfig, err := cfg.GetExchangeConfig("Coinbene")
+	if err != nil {
+		log.Fatalf("Test Failed - Coinbene Setup() init error: %v", err)
+	}
+	coinbeneConfig.Websocket = true
+	coinbeneConfig.AuthenticatedAPISupport = true
+	coinbeneConfig.APISecret = testAPISecret
+	coinbeneConfig.APIKey = testAPIKey
+	c.Setup(&coinbeneConfig)
+
+	os.Exit(m.Run())
+}
+
+func areTestAPIKeysSet() bool {
+	if c.APIKey != "" && c.APIKey != "Key" &&
+		c.APISecret != "" && c.APISecret != "Secret" {
+		return true
+	}
+	return false
+}
+
+func TestFetchTicker(t *testing.T) {
+	t.Parallel()
+	_, err := c.FetchTicker(btcusdt)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestFetchOrderbooks(t *testing.T) {
+	t.Parallel()
+	_, err := c.FetchOrderbooks(btcusdt, 100)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetTrades(t *testing.T) {
+	t.Parallel()
+	_, err := c.GetTrades(btcusdt)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetAllPairs(t *testing.T) {
+	t.Parallel()
+	_, err := c.GetAllPairs()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetPairInfo(t *testing.T) {
+	t.Parallel()
+	_, err := c.GetPairInfo(btcusdt)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetUserBalance(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip("API keys required but not set, skipping test")
+	}
+	_, err := c.GetUserBalance()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestPlaceOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip("skipping test, either api keys or manipulaterealorders isnt set correctly")
+	}
+	_, err := c.PlaceOrder(140, 1, btcusdt, "1", "")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestFetchOrderInfo(t *testing.T) {
```

Comment:
I am expecting an error here. 
JSON raw response with API keys set = 
```
{"code":200,"message":null,"data":{"orderId":null,"baseAsset":null,"quoteAsset":null,"orderDirection":null,"quantity":null,"filledQuantity":null,"amount":null,"filledAmount":null,"avgPrice":null,"orderStatus":null,"orderTime":null,"updateTime":null,"fee":null,"takerFeeRate":null,"makerFeeRate":null,"symbol":null,"orderPrice":null}}
```

You can add in a check prior to returning to see if orderID == 0 and return an error.New("order id cannot be found")

Then do a pass over the tests again especially API tests just to ensure they are not sending dummy payloads back when they cant find anything on their database side.